Member.java

```diff
@@ -0,0 +1,83 @@
+package kadai02;
+
+import java.time.LocalDate;
+
+public class Member {
+    private String name;                    //会員名
+    private int age;                        //年齢
+    private LocalDate joinDate;             //入会日
+    private boolean withdrawal;             //退会　退会の時にture
+    private LocalDate withdrawalDate;       //退会日
+
+    //コンストラクターにて、名前、年齢、入会日を初期設定。退会フラグはfalse、退会日はnullを代入
+    public Member(String name, int age, LocalDate joinDate) {
+        this.name = name;
+        this.age = age;
+        this.joinDate = joinDate;
+        withdrawal = false;         //入会時はfalse
+        withdrawalDate = null;      //入会時はnull
+    }
+
+    //会員の会員名、年齢、入会日を表示するメソッド
+    public void showProfile() {
+        System.out.println("会員名：" + getName());
+        System.out.println("年齢　：" + getAge());
+        System.out.println("入会日:" + getJoinDate());
+        System.out.println();
+    }
+
+    //入会中であるかどうかをboolean型で返すメソッド。入会中の時true。
+    //入会中条件：入会日が今日より過去かつ、退会フラグがfalse。
+    public boolean isActive() {
+        LocalDate today = LocalDate.now();
+        boolean isActiveBoolean;
+
+        if ((joinDate.compareTo(today) <= 0) && (withdrawal == false)) {
+            isActiveBoolean = true;
+        } else {
+            isActiveBoolean = false;
+        }
+        return isActiveBoolean;
```

Comment:
短いのは正義ではないですが、こういう書き方もあります。
```suggestion
        LocalDate today = LocalDate.now();
        return joinDate.compareTo(today) <= 0 && withdrawal == false
```
```suggestion
        LocalDate today = LocalDate.now();
        return (joinDate.equals(today) || joinDate.before(today) && !withdrawal
```
```suggestion
        LocalDate today = LocalDate.now();
        return joinDate.compareTo(today) <= 0 && !withdrawal
```

Reply:
test
対応しました！
ご確認お願いします！
383e86461c66d6d0f87ba159dc5c703e181f5b5a

Reply:
これはもうびっくりです。そもそもif文いらないんじゃない。かということですね。
returnに条件直接書いて終わりですね。いやー。まだまだ修行がいるなぁと思うと同時に、非常に、面白いです！
第4回の過去授業動画を見ておりまして、プログラマーはコードを書くけども、できるだけ書かない方が良い。
if文はバグの素。であることをご説明されており、納得した次第です。ありがとうございます。
提案変更を受けました。

Reply:
> プログラマーはコードを書くけども、できるだけ書かない方が良い。
if文はバグの素。であることをご説明されており、納得した次第です。

林さんは学習スピードが異常に早くていつも驚いています！
どんどんプログラマーになっていっている・・・！